src/libponyrt/actor/actor.c

```diff
@@ -432,69 +438,124 @@ bool ponyint_actor_run(pony_ctx_t* ctx, pony_actor_t* actor, bool polling)
 
   }
 
-  // If we mark the queue as empty, then it is no longer safe to do any
-  // operations on this actor that aren't concurrency safe unless, the actor
-  // has an rc of 0 and the cycle detector isn't aware of the actor's
-  // existence.
-  //
-  // If there are other actors with references to this actor then once its
-  // queue is marked as empty, it can be rescheduled on another scheduler
-  // thread. "Other actors" includes the cycle detector if it is running, so...
-  // beware of any code in this function after marking the queue as empty.
-  // Getting it wrong will result in hard to track down segfaults.
-  bool empty = ponyint_messageq_markempty(&actor->q);
-
-  if (empty && (actor->gc.rc == 0) && has_flag(actor, FLAG_BLOCKED))
+  if ((actor->gc.rc == 0) && has_flag(actor, FLAG_BLOCKED))
   {
     // Here, we is what we know to be true:
     //
-    // - the actor has no messages in its queue
+    // - the actor is blocked and likely has no meesages in its queue
```

Comment:
```suggestion
    // - the actor is blocked and likely has no messages in its queue
```

Reply:
Given this is a list of things, I think it should be split into 2 items at the "and"

Reply:
addressed and pushing soon.